src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,57 +1,137 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+import { Movie } from '../../types/Movie';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+import { MovieData } from '../../types/MovieData';
+import { normalizeMovieData } from '../utils/Normalize';
+
+ type Props = {
+   addMovie: Dispatch<SetStateAction<Movie[]>>
+ };
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isMovieExist, setIsMovieExist] = useState(false);
+
+  const handleInputChange = (event:React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setIsError(false);
+    setMovie(null);
+    setIsMovieExist(false);
+  };
+
+  const handleSubmitForm = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    try {
+      const fetchMovie = await getMovie(query);
+
+      if ((fetchMovie as MovieData).imdbID) {
+        setMovie(normalizeMovieData(fetchMovie as MovieData));
+      } else {
+        setIsError(true);
+      }
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleAddMovie = (newMovie: Movie) => {
+    addMovie(prevMovies => {
+      if (prevMovies.some(
```

Comment:
Consider extracting this function to a separate utility function to make it reusable and easier to test.